src/keys.rs

```diff
@@ -256,6 +269,12 @@ impl From<&SpendingKey> for CommitIvkRandomness {
     }
 }
 
+impl From<&InternalSpendingKey> for CommitIvkRandomness {
+    fn from(sk: &InternalSpendingKey) -> Self {
+        CommitIvkRandomness(to_scalar(PrfExpand::OrchardRivkInternal.expand(&sk.0)))
```

Comment:
This isn't correct. rivk<sub>internal</sub> is derived from ak, nk, and rivk of the external full viewing key, not directly from the spending key.